### PR TITLE
Refactor webinterface

### PR DESCRIPTION
**Pull request by RobbinBouwmeester**

### CODE REVIEW

File: webinterface/pages/DDA_Quant_ion.py [hunk 1]
```diff
@@ -99,617 +90,535 @@ def generate_input_field(self, input_format: str, content: dict):
...
-                if META_FILE_UPLOADER_UUID in st.session_state.keys():
-                    del st.session_state[META_FILE_UPLOADER_UUID]
-                if COMMENTS_SUBMISSION_UUID in st.session_state.keys():
-                    del st.session_state[COMMENTS_SUBMISSION_UUID]
-                if CHECK_SUBMISSION_UUID in st.session_state.keys():
-                    del st.session_state[CHECK_SUBMISSION_UUID]
-                if BUTTON_SUBMISSION_UUID in st.session_state.keys():
-                    del st.session_state[BUTTON_SUBMISSION_UUID]
-
+                if self.variables_dda_quant.meta_file_uploader_uuid in st.session_state.keys():
```
Comment (RobbinBouwmeester) [inline webinterface/pages/DDA_Quant_ion.py:135]: * maybe not call dda_quant

File: webinterface/pages/DDA_Quant_ion.py [hunk 2]
```diff
@@ -99,617 +90,535 @@ def generate_input_field(self, input_format: str, content: dict):
...
-            st.session_state[FIG_METRIC].data[0].y = fig_metric.data[0].y
+        st.session_state[self.variables_dda_quant.check_submission] = st.checkbox(
+            "I confirm that the metadata is correct",
+            key=self.variables_dda_quant.check_submission_uuid,
+        )
 
-            st.session_state[PLACEHOLDER_FIG_COMPARE].plotly_chart(
-                st.session_state[FIG_METRIC], use_container_width=True
-            )
+    def create_sample_name(self) -> str:
```
Comment (RobbinBouwmeester) [inline webinterface/pages/DDA_Quant_ion.py:435]: Check datapoint class, possible duplication

**Comment by mlocardpaulet** [comment]
Hi! 

I wanted to try out the PR version (maybe it is too early, but I was curious), so I tried it out locally. 
When the page load (without me doing anything), the figure of the module is different. I mean... One point moves. It is the last point updated (Proline_2.1_20240318_162947). Any idea why this would happen?
Here are the two figures (local version corresponding to the PR, and sever version).

local_version_20240405
server_version_20240405

File: proteobench/utils/plotting/plot.py [hunk 3]
```diff
@@ -66,7 +66,6 @@ def plot_fold_change_histogram(result_df: pd.DataFrame, species_ratio: dict) ->
     @staticmethod
     def plot_metric(
         benchmark_metrics_df: pd.DataFrame,
-        reinitialize_table: bool = False,
         software_colors: dict = {
```
Comment (mlocardpaulet) [inline proteobench/utils/plotting/plot.py:69]: not for now, but I think that we should consider having the software colors defined somewhere else. Maybe in the module .toml?

File: proteobench/utils/quant_datapoint.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    if isinstance(list(row.keys())[0], str):
+        min_quant = str(min_quant)
+    if isinstance(row, dict) and min_quant in row and isinstance(row[min_quant], dict):
+        return row[min_quant].get("nr_prec")
+    return None
+
+
+@dataclass
+class Datapoint:
+    """Data used to stored the"""
```
Comment (mlocardpaulet) [inline proteobench/utils/quant_datapoint.py:31]: this docstring is broken, isn't it?

File: proteobench/utils/quant_datapoint.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    enzyme: str = None
+    allowed_miscleavages: int = 0
+    min_peptide_length: int = 0
+    max_peptide_length: int = 0
+    is_temporary: bool = True
+    intermediate_hash: str = ""
+    results: dict = None
+    median_abs_epsilon: int = 0
+    nr_prec: int = 0
+    comments: str = ""
```
Comment (mlocardpaulet) [inline proteobench/utils/quant_datapoint.py:54]: Is this the "comments_for_submission"? Or the comments that can manually be added when uploaded the MS file? If it is the latest, should we really keep it? Isn't it confusing to have two comments fields?

File: webinterface/pages/DDA_Quant_ion.py [hunk 6]
```diff
@@ -99,617 +90,535 @@ def generate_input_field(self, input_format: str, content: dict):
...
+        st.markdown(open("pages/markdown_files/DDA_Quant_ion/file_description.md", "r").read(), unsafe_allow_html=True)
         st.header("Input and configuration")
-
-        st.markdown(
-            """
-                   Scroll down if you want to see the public benchmark runs publicly available
-                   today.
-                    """
-        )
+        st.markdown(self.texts.ShortMessages.initial_results)
```
Comment (mlocardpaulet) [inline webinterface/pages/DDA_Quant_ion.py:107]: I don't know why, here there is a `"` printed before the sentence in the webinterface. 
image

**Comment by mlocardpaulet** [comment]
I uploaded a MQ evidence file, it worked without error, but the point is not printed (or not bigger). I have to change the minimum number of samples to see it appear big. 
Then I changed the slider value again and I got an error message:
```
TypeError: 'NoneType' object is not subscriptable
Traceback:
File "/Users/locard/miniconda3/envs/ProteoBench/lib/python3.11/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 530, in _run_script
    self._session_state.on_script_will_rerun(rerun_data.widget_states)
File "/Users/locard/miniconda3/envs/ProteoBench/lib/python3.11/site-packages/streamlit/runtime/state/safe_session_state.py", line 61, in on_script_will_rerun
    self._state.on_script_will_rerun(latest_widget_states)
File "/Users/locard/miniconda3/envs/ProteoBench/lib/python3.11/site-packages/streamlit/runtime/state/session_state.py", line 500, in on_script_will_rerun
    self._call_callbacks()
File "/Users/locard/miniconda3/envs/ProteoBench/lib/python3.11/site-packages/streamlit/runtime/state/session_state.py", line 513, in _call_callbacks
    self._new_widget_state.call_callback(wid)
File "/Users/locard/miniconda3/envs/ProteoBench/lib/python3.11/site-packages/streamlit/runtime/state/session_state.py", line 260, in call_callback
    callback(*args, **kwargs)
File "/Users/locard/Documents/GitHub/ProteoBench/webinterface/pages/DDA_Quant_ion.py", line 292, in slider_callback
    st.session_state[self.variables_dda_quant.all_datapoints] = self.ionmodule.filter_data_point(
                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/locard/miniconda3/envs/ProteoBench/lib/python3.11/site-packages/proteobench/modules/dda_quant_base/module.py", line 83, in filter_data_point
    filter_df_numquant_median_abs_epsilon(v, min_quant=default_val_slider) for v in all_datapoints["results"]
                                                                                    ~~~~~~~~~~~~~~^^^^^^^^^^^
```

**Comment by mlocardpaulet** [comment]
I also have an error when I try to upload a Proline file: 
```
TypeError: 'dict' object is not callable
Traceback:
File "/Users/locard/Documents/GitHub/ProteoBench/webinterface/pages/DDA_Quant_ion.py", line 234, in _run_proteobench
    result_performance, all_datapoints, input_df = self.ionmodule.benchmarking(
                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/Users/locard/miniconda3/envs/ProteoBench/lib/python3.11/site-packages/proteobench/modules/dda_quant_ion/module.py", line 42, in benchmarking
    self.precursor_name, parse_settings.species_expected_ratio(), parse_settings.species_dict()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: proteobench/io/data/__init__.py
Comment (vedran-kasalica) [inline proteobench/io/data/__init__.py:1]: Is there a reason to have an empty folder with an empty `__initi__` file? If the goal is to track an empty folder a better practice is to add `.gitkeep`. Some people add `.gitignore` instead.

Comment (RobbinBouwmeester) [reply proteobench/io/data/__init__.py:1]: Will change to .gitkeep

File: webinterface/pages/markdown_files/__init__.py
Comment (vedran-kasalica) [inline webinterface/pages/markdown_files/__init__.py:1]: There is probably no need to have the `init` file in the folders under `markdown_files`

**Comment by vedran-kasalica** [comment]
I got the same error (`'NoneType' object is not subscriptable`) as @mlocardpaulet  while testing. The error occurs on the `main` branch as well. I would say that it is not PR-related, so I created an issue for it https://github.com/Proteobench/ProteoBench/issues/286. 

The changes look good to me! For future development, we might reconsider placing an `__init__.py` file in every folder, as this practice implicitly designates each directory as a distinct Python package. This is particularly unnecessary in folders that do not contain any Python files.


*I didn't test the web interface by uploading new data.

File: webinterface/pages/DDA_Quant_ion.py [hunk 7]
```diff
@@ -187,528 +121,524 @@ def _main_page(self):
...
             )
 
-        if "slider_id" in st.session_state.keys():
-            default_val_slider = st.session_state[st.session_state["slider_id"]]
-        else:
-            default_val_slider = DEFAULT_VAL_SLIDER
+            if not pr_url:
+                del st.session_state[self.variables_dda_quant.submit]
+            else:
+                id = str(
```
Comment (enryH) [inline webinterface/pages/DDA_Quant_ion.py:361]: @RobbinBouwmeester @wolski I get the error here. #288

**Pull request merged**